### ISSUE TITLE
How to start docker container with local image ?

### ISSUE THREAD

**Issue by jonasv** [jonnyzzz/TeamCity.Virtual#49]
How do I start docker container with local image ?

For example I have got this at the moment:
[root@tcagent docker]# docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
image             tag                 6584e9c5cc79        44 minutes ago      1.347 GB

How can I start image/tag ? I been trying to get this working for the past 30minutes, no joy! 

Would be great to have more examples how to use (and especially build) this plugin >>Author

thanks

**Comment by jonasv** [comment]
Ok, looks like image is now found, but I still get failed build because your plugin is trying to pull image from the repo first and of course image is not there. How I can I disable pull'ing ?

**Comment by jonasv** [comment]
I think is a bug. Plugin first trying to pull image from the repo and this step fails, then it will check local image and image is found and status is 0, but whole task fails because plugin initially failed to find image in the repo ? I'm using docker directly without vagrant.

Author, can you please confirm ?

**Comment by jonnyzzz** [comment]
Yes, we first try to pull the image (to make sure it's the latest image).
I was wondering if there is a rule, the plugin could use to tell local
image from public image? The other option is to add an explicit checkbox to
control weather pull is executed or not
​

**Comment by jonasv** [comment]
How can I prevent image get pulled at the beginning ? There is NO reason to pull anything in my case because image is already in file system!

**Comment by jonnyzzz** [comment]
Sounds like a missing feature for the plugin, thanks for reporting it. It'd
be nice if you are able to submit a pull request for this

Best regards,
Eugene Petrenko

**Comment by jonasv** [comment]
I can't, because I don't code in java sorry. Anyway, Jonnyzzz I still can't get your plugin working, previously  I was trying to run it directly from docker, looks like it is not supported with local image... Now I rearranged everything to run via vagrant but still can't get it working.


[14:24:13][Step 6/6] vagrant: Running the script using /bin/bash (2s)

I'm not sure why this step is being called ? I don't have it anywhere.

Could you please explain how to run your plugin ? Looks like I'm not the only one who is facing issues start using your work.

thanks

**Comment by jonasv** [comment]
Also, is ssh mandatory inside container to make your plugin work ?

**Comment by jonnyzzz** [comment]
No, ssh is not needed. Only bash.

Best regards,
Eugene Petrenko

**Comment by jonasv** [comment]
[14:52:10][Step 6/6] vagrant: Running the script using /bin/bash (2s)
[14:52:10][vagrant] Executing command: **vagrant ssh -c** "/bin/bash -c 'cd /checkout && . teamcity7280727771818780326.build.cmd'"

So why is your plugin then trying to run this code ?

**Comment by jonnyzzz** [comment]
I see what you mean. Yes for Vagrant we do need ssh, but as far as I know
all Vagrant images were required to have ssh running inside, so Vagrant is
able to connect there.

Please note, we actually call `vagrant ssh` command (not directly an ssh
command, but vagrant calls ssh for us in most of the cases)
​

**Comment by jonasv** [comment]
Jonnyzzz, is there a way to speak with you in real time ? For example irc somewhere .. ?

**Comment by jonnyzzz** [comment]
Let's setup a public chatroom for the plugin codebase. I'm ok to use any
chat that works from browser. Maybe Gitter? Do you use something similar
already? Is it ok to you to make a PR to include this chat link/icon into
our README.md?​

**Comment by jonasv** [comment]
Hey,

I sent you invite to gitter , I'm waiting for you there already. Once you join then we can discuss.

Thanks

**Comment by jonasv** [comment]
Just in case you can't join, "tcvirtual/Lobby" is the name of the room.

Cheers

**Comment by jonasv** [comment]
Mate, you joining .. ?

**Comment by jonasv** [comment]
Hey,

could you please respond if everything is ok ?

Thanks

**Comment by maxgorovenko** [comment]
I have the same requirement. Is there any progress?
